src/elm/Main.elm

```diff
@@ -374,7 +375,7 @@ type Msg
     | ShowHideHelp (Maybe Bool)
     | ShowHideIdentity (Maybe Bool)
     | Copy String
-    | DownloadFile String String String
+    | DownloadFile String (String -> String) String String
```

Comment:
adding a string transformation function so the download Msg can be supplied an expensive function that is only called when the user executes the code (clicks the button)